docs/content/docs/concepts/summarization.md

```diff
@@ -0,0 +1,101 @@
+-   A summarizer client is like any other client connected to the document except that users cannot interact with this client, and it only works on the state it receives from other clients. It has a brand-new container with its own connection to services.
+
+> Note that if the summarizer client closes, the "summary client election" process will choose a new one, if applicable. The default "summary client election" algorithm is to select the oldest "write" client as the parent summarizer client which in turn will create the summarizer client.
+
+## When are summaries generated?
+
+The summarizer client periodically generates summary based on heuristics calculated based on configurations such as the number of user or system ops received, the amount of time a client has been idle (hasn't received any ops), the maximum time since last summary, maximum number of ops since last summary, etc. The heuristic configurations are defined by an _ISummaryConfigurationHeuristics_ interface defined in [this file](../../src/containerRuntime.ts).
+
+The summarizer client uses a default set of configurations defined by _DefaultSummaryConfiguration_ in [this file](../../src/containerRuntime.ts). These can be overridden by providing a new set of configurations as part of container runtime options during creation.
+
+## How are summaries generated?
+
+When summarization process is triggered, every object in the container's object tree that has data to be summarized is asked to generate its summary, starting at the container runtime which is at the root. There are various objects that participate in the summary process and generate its summary such as data stores, DDSes, garbage collector, blob manager, id compressor, etc. Note that the user data is in the DDSes.
+
+### Summary Lifecycle
+
+The lifecycle of a summary starts when a "parent summarizer" client is elected.
+
+-   The parent summarizer spawns a non-interactive summarizer client.
+-   The summarizer client periodically starts a summary as per heuristics. A summary happens at a particular sequence number called the "summary sequence number" or reference sequence number for the summary.
+-   The container runtime (or simply runtime) generates a summary tree (described in the ["What does a summary look like?"](#what-does-a-summary-look-like) section below).
+-   The runtime uploads the summary tree to the Fluid storage service which returns a handle (unique id) to the summary if the upload is successful. Otherwise, it returns a failure.
+-   The runtime submits a "summarize" op to the Fluid ordering service containing the uploaded summary handle and the summary sequence number.
+-   The ordering service stamps it with a sequence number (like any other op) and broadcasts the summarize op.
```

Comment:
The summarize op isn't really for other clients. It's part of a 2-commit summary process:
- Summarizer uploads a summary and the server returns a handle (basically a unique id) but the summary isn't final yet meaning that the server hasn't committed it yet.
- The client sends a summary op with the uploaded handle so that the server can identify which summary this op is for. It then augments the uploaded summary with some info, generates a new handle and sends this handle in a summary ack.
- Once the summarizer clients receives the summary ack, it knows that the summary is now accepted and committed by the server.

Reply:
Should I mention this or part of it to make the purpose of summary op clear?

Reply:
Updated the section to include the purpose of the summary broadcast and ack / nack.